src/core/kernel.cpp

```diff
@@ -222,7 +222,7 @@ namespace occa {
 
     // Add defines
     for (const auto &entry : props["defines"].object()) {
-      if (entry.second.isString()) {
```

Comment:
Can you revert to what the logic was before? 
```cpp
      kernelHeader += "#define ";
      kernelHeader += ' ';
      kernelHeader += entry.first;
      kernelHeader += ' ';
      kernelHeader += (std::string) entry.second;
      kernelHeader += '\n';
```

Or we can also check for `isBool()` and `isNull()`. I don't think we want to print out object or array types

Reply:
How are booleans and nulls rendered as strings - will that be something that makes sense in a `#define`?
I can either remove the check completely, reverting to the previous logic, or add the full quartet, your call.

Reply:
Booleans will get stringified as `true` or `false`, and the null case as `NULL`. I like your approach since we explicitly limit the types we want to `#define`.

Reply:
Thanks for making the change, will merge after checks run 🎉